### PR TITLE
fix(#2415): nvim 0.8 highlight overhaul support, limited to only show highest highlight precedence

### PR DESCRIPTION
**Pull request by Confidenceman02**
nvim-tree is using `nvim_get_hl` which was introduced in nvim 0.9 to replace the unstable `get_hl_defs` in the following [commit](https://github.com/neovim/neovim/pull/22693/files).

Unfortunately this raises an error in 0.8 nvim versions due to the function not existing.

```
Failed to run `config` for nvim-tree.lua
...are/nvim/lazy/nvim-tree.lua/lua/nvim-tree/appearance.lua:199: attempt to call field 'nvim_get_hl' (a nil value)
stacktrace:
  - ~/.config/nvim/lua/confidenceman02/plugins/nvim-tree.lua:14 _in_ **config**
  - ~/.config/nvim/lua/confidenceman02/lazy.lua:14
```

- Fall back to get_hl_defs when detecting 0.8
- Set the 'link' property to nil to emulate `link = false` in `builder.lua`

### CODE REVIEW

File: lua/nvim-tree/renderer/builder.lua [hunk 1]
```diff
@@ -266,7 +266,15 @@ function Builder:create_combined_group(groups)
 
     -- build the highlight, overriding values
     for _, group in ipairs(groups) do
-      local hl = vim.api.nvim_get_hl(0, { name = group, link = false })
+      local hl
+      if vim.fn.has "nvim-0.8" == 1 then
+        hl = vim.api.nvim__get_hl_defs(0)[group] or {}
+        if hl["link"] then
```
Comment (Confidenceman02) [inline lua/nvim-tree/renderer/builder.lua:272]: This check feels a little overkill considering we default to a table. I'm too much of a lua noob to know for sure.

Comment (alex-courtis) [reply lua/nvim-tree/renderer/builder.lua:272]: That's fine; safety first.

File: lua/nvim-tree/appearance.lua [hunk 2]
```diff
@@ -196,8 +196,15 @@ function M.setup()
 
   -- hard link override when legacy only is present
   for from, to in pairs(LEGACY_LINKS) do
-    local hl_from = vim.api.nvim_get_hl(0, { name = from })
-    local hl_to = vim.api.nvim_get_hl(0, { name = to })
+    local hl_from
+    local hl_to
+    if vim.fn.has "nvim-0.8" == 1 then
```
Comment (alex-courtis) [inline lua/nvim-tree/appearance.lua:201]: Unfortunately we'll have to invert this condition: `nvim-0.8` is true when using 0.9

I discovered this as `nvim__get_hl_defs` has been completely removed from 0.9. So much for non-breaking changes...

**Review by alex-courtis — CHANGES_REQUESTED** [review]
Many thanks for your work!

Let's invert those conditions, you test on 0.8, I test on 0.9 and we're ready to go.

File: lua/nvim-tree/renderer/builder.lua [hunk 3]
```diff
@@ -295,16 +295,24 @@ function Builder:add_highlights(node)
...
   end
 
-  -- one or many name groups
+  -- one or many name groups; <= 0.8 always uses highest due to lack of a practical nvim_get_hl equivalent
   if #name_groups > 1 then
-    name_hl_group = self:create_combined_group(name_groups)
+    if vim.fn.has "nvim-0.9" == 1 then
+      name_hl_group = self:create_combined_group(name_groups)
+    else
+      name_hl_group = name_groups[#name_groups]
```
Comment (alex-courtis) [inline lua/nvim-tree/renderer/builder.lua:314]: There is no practical API to follow links all the way to the actual definitions, so we don't try.

**Comment by alex-courtis** [comment]
Apologies for clobbering your changes; this is on the urgent side.

**Review by alex-courtis — APPROVED** [review]

**Comment by Confidenceman02** [comment]
> Apologies for clobbering your changes; this is on the urgent side.
> 
> Many thanks for raising and fixing this!

Ah nice one! 

Yea sorry I was on an extended long, long weekend!

Thanks again.

**Pull request merged**